src/Admin/AbstractAdmin.php

```diff
@@ -3410,6 +3411,7 @@ protected function attachInlineValidator()
 
         // add the custom inline validation option
         $metadata = $this->validator->getMetadataFor($this->getClass());
+        \assert($metadata instanceof GenericMetadata);
```

Comment:
What if it's not a `GenericMetadata` ? Is it possible ?

Reply:
I agree that this method cannot work if the validator return something else.

I meant. The `setValidator` method is written this way
```
public function setValidator($validator)
    {
        // NEXT_MAJOR: Move ValidatorInterface check to method signature
        if (!$validator instanceof ValidatorInterface) {
            throw new \InvalidArgumentException(
                'Argument 1 must be an instance of Symfony\Component\Validator\Validator\ValidatorInterface'
            );
        }

        $this->validator = $validator;
    }
```

So theorically the developer could provide a validator which return something else than a `GenericMetada`. When calling the `setValidator` he'll received no error.

But then, he will receive an error in this method if the validator doesn't return a `GenericMetadata`.
I don't know how validator and metadata works, but if this method can return something else, it's more helpful to write
```
if (!$metadata instanceof GenericMetadata) {
    throw new \RuntimeException('Some explicit message about the problem and how to fix it.');
}
```
Instead of `method not found` or the error thrown by `assert`.

IMHO, assert should be used when we know something like
```
$extension = $this->twig->getExtension(SonataAdminExtension::class);
\assert($extension instanceof SonataAdminExtension);
```
and not when we need something like here.

Reply:
to be honest, I don't fully understand where I should provide `assert` and where throw exception explicitly.
Point me at somewhere where I can get more info on it